src/ParseFile.js

```diff
@@ -292,6 +300,52 @@ class ParseFile {
     );
   }
 
+  /**
+   * Sets metadata to be saved with file object. Overwrites existing metadata
+   * @param {Object} metadata Key value pairs to be stored with file object
+   */
+  setMetadata(metadata: Object) {
+    if (metadata && typeof metadata === 'object') {
+      Object.keys(metadata).forEach((key) => {
+        this.addMetadata(key, metadata[key]);
+      });
+    }
+  }
+
+  /**
+   * Sets metadata to be saved with file object. Adds to existing metadata
+   * @param {String} key
+   * @param {Mixed} value
+   */
+  addMetadata(key: String, value: any) {
+    if (typeof key === 'string') {
+      this._metadata[key] = value;
+    }
+  }
+
+  /**
+   * Sets tags to be saved with file object. Overwrites existing tags
+   * @param {Object} tags Key value pairs to be stored with file object
+   */
+  setTags(tags: Object) {
+    if (tags && typeof tags === 'object') {
+      Object.keys(tags).forEach((key) => {
+        this.addTag(key, tags[key]);
+      });
+    }
+  }
+
+  /**
+   * Sets tags to be saved with file object. Adds to existing tags
+   * @param {String} key
+   * @param {Mixed} value
+   */
+  addTag(key: String, value: String) {
+    if (typeof key === 'string') {
+      this._tags[key] = value;
+    }
+  }
+
   static fromJSON(obj): ParseFile {
```

Comment:
Can you update fromJSON with _metadata and _tag? 

Just realized type was missing. I think we can skip it for now.

Reply:
I had this added at first but it broke a bunch of tests. Doing this would cause the metadata and tags to be saved to mongo. I believe leaving metadata and tags out makes sense because the they are both stored in S3 which should be the source of truth. So if you want the metadata or tags from a file that has already been saved, it should come from S3 (or whatever storage provider they're using)